server/datastore/mysql/hosts.go

```diff
@@ -1615,3 +1615,115 @@ func (ds *Datastore) UpdateHost(ctx context.Context, host *fleet.Host) error {
 	}
 	return nil
 }
+
+// OSVersions gets the aggregated os version host counts. If a non-nil teamID is passed, it will filter hosts by team.
+func (ds *Datastore) OSVersions(ctx context.Context, teamID *uint, platform *string) (*fleet.OSVersions, error) {
+	query := `
+SELECT
+    json_value,
+    updated_at
+FROM aggregated_stats
+WHERE
+    id = ? AND
+    type = 'os_versions'
+`
+
+	var row struct {
+		JSONValue *json.RawMessage `db:"json_value"`
+		UpdatedAt time.Time        `db:"updated_at"`
+	}
+
+	var args []interface{}
+	if teamID == nil { // all hosts
+		args = append(args, 0)
+	} else {
+		args = append(args, *teamID)
+	}
+
+	if err := sqlx.GetContext(ctx, ds.reader, &row, query, args...); err != nil {
+		return nil, err
+	}
+
+	osVersions := &fleet.OSVersions{
+		CountsUpdatedAt: row.UpdatedAt,
+	}
+
+	if row.JSONValue != nil {
+		err := json.Unmarshal(*row.JSONValue, &osVersions.OSVersions)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// filter by os versions by platform
+	if platform != nil {
+		var filtered []fleet.OSVersion
+		for _, osVersion := range osVersions.OSVersions {
+			if *platform == osVersion.Platform {
```

Comment:
Should that support "meta-platforms"? I.e. should "linux" include "rhel", "debian", etc.? I see on the ticket that it is macos only for now, so that may be fine, but you should probably leave a note for future development that either here or in UpdateOSVersions we'll need to consider the "platform group" (or however we should call this, only "linux" for now I believe, and we'd probably want to store counts by "granular platform" like "rhel" and "debian" in aggregated_stats, as we can aggregate those programatically if we want counts for all "linux", but we can't do the other way around if we only store counts for "linux").

Reply:
Currently, the platforms are taken from the hosts table as is. While the ticket only mentions darwin, you can filter by debian, rhel, etc.

Reply:
Maybe @noahtalerman can answer the question about platform groups. I feel like it would be a reasonable request, but I think we should also return that in other places eg hosts for consistency.

Reply:
We do have that in some other places, if you look for uses `fleet.ExpandPlatform` and `fleet.PlatformFromHost`. But yeah, Noah can clarify if this is needed (and in any case it's not for that first iteration).

Reply:
Ah given that it is platform + os version, it might not make sense to group by "linux" anyway (unless in this case "os version" is "kernel version").